### DISCUSSION TITLE
Is it really necessary to run ztunnel/cni DaemonSets on ALL nodes?

### DISCUSSION THREAD

**Discussion by m8-t** [istio/istio#54152] [Q&A] [answered]
Title is the question, but a little bit more background: I understand that DaemonSets are meant to run on any node but for good reasons the Kubernetes masters and thus other nodes (storage, dedicated ingress controllers etc) are not part of the mesh - so is there a con to let ztunnel and istio-cni pods run only on nodes that will be part of the data-plane (worker nodes)? Just curious...and in case of a bug or something I do not have issues with other nodes besides worker nodes. I saw there is a `nodeSelector` in  the DaemonSet template for ztunnel (not stated in values, but can be overridden) but not for istio-cni (affinity only).

**Comment by howardjohn** [comment, marked as answer]
No, that would work fine - there is just a risk you may accidentally not run it on a node that does need it and unexpectedly have a pod out of the mesh when I should be in the mesh
